### ISSUE TITLE
com.android.dex.DexException when reading Parcelable in SdlRouterService.onReceive()

### ISSUE THREAD

**Issue by askirk** [smartdevicelink/sdl_android#393]
Stacktrace:
```
Fatal Exception: java.lang.RuntimeException: Error receiving broadcast Intent { act=com.sdl.android.newservice flg=0x10 (has extras) } in com.smartdevicelink.transport.SdlRouterService$2@1eb4c767
       at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:943)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5438)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:917)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:712)
Caused by com.android.dex.DexException: Declared length 11 doesn't match decoded length of 0
       at com.android.dex.Dex$Section.readString(Dex.java:594)
       at com.android.dex.Dex$StringTable.get(Dex.java:893)
       at com.android.dex.Dex$StringTable.get(Dex.java:890)
       at java.lang.Class.getDexCacheString(Class.java:461)
       at java.lang.reflect.ArtField.getName(ArtField.java:77)
       at java.lang.Class.getDeclaredFieldInternal(Class.java:944)
       at java.lang.Class.getPublicFieldRecursive(Class.java:1058)
       at java.lang.Class.getField(Class.java:1046)
       at android.os.Parcel.readParcelableCreator(Parcel.java:2277)
       at android.os.Parcel.readParcelable(Parcel.java:2239)
       at com.smartdevicelink.transport.SdlRouterService$LocalRouterService.(SourceFile)
       at com.smartdevicelink.transport.SdlRouterService$LocalRouterService$1.createFromParcel(SourceFile:4045)
       at android.os.Parcel.readParcelable(Parcel.java:2246)
       at android.os.Parcel.readValue(Parcel.java:2146)
       at android.os.Parcel.readArrayMapInternal(Parcel.java:2479)
       at android.os.BaseBundle.unparcel(BaseBundle.java:221)
       at android.os.Bundle.getParcelable(Bundle.java:804)
       at android.content.Intent.getParcelableExtra(Intent.java:4841)
       at com.smartdevicelink.transport.SdlRouterService$2.onReceive(SourceFile:223)
```

Occurs at https://github.com/smartdevicelink/sdl_android/blob/master/sdl_android_lib/src/com/smartdevicelink/transport/SdlRouterService.java#L2025

**Comment by askirk** [comment]
Try/catch and set name or launchIntent to null if we get an NPE.

**Comment by askirk** [comment]
![bitmoji](https://render.bitstrips.com/v2/cpanel/10216702-187486277_14-s1-v1.png?transparent=1&palette=1&width=246)

**Comment by joeygrover** [comment]
Fixed with #412
